src/sentry/snuba/metrics/datasource.py

```diff
@@ -440,19 +462,28 @@ def get_single_metric_info(
 
 
 def get_tags(
-    projects: Sequence[Project], metric_names: Optional[Sequence[str]], use_case_id: UseCaseKey
+    projects: Sequence[Project], metrics: Optional[Sequence[str]], use_case_id: UseCaseKey
 ) -> Sequence[Tag]:
     """Get all metric tags for the given projects and metric_names"""
     assert projects
 
     try:
-        tags, _ = _fetch_tags_or_values_per_ids(
-            projects=projects,
-            metric_names=metric_names,
-            column="tags.key",
-            referrer="snuba.metrics.meta.get_tags",
-            use_case_id=use_case_id,
-        )
+        if len(metrics) and MRI_SCHEMA_REGEX.match(metrics[0]):
```

Comment:
Shouldn't we verify that we have all the metrics as MRIs?

Reply:
Yep, done!